dotCMS/src/main/java/com/dotcms/util/pagination/ThemePaginator.java

```diff
@@ -1,75 +1,124 @@
+import com.dotmarketing.portlets.folders.model.Folder;
 import com.dotmarketing.util.PaginatedArrayList;
+import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.model.User;
-import org.elasticsearch.search.SearchHit;
-import org.elasticsearch.search.SearchHits;
 
-import static com.dotcms.util.CollectionsUtils.map;
-import static org.mockito.Mockito.when;
 
+import com.liferay.util.StringPool;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 /**
  * Handle theme pagination
  */
-public class ThemePaginator implements Paginator<Contentlet> {
+public class ThemePaginator implements Paginator<Folder> {
 
-    public final static String HOST_ID_PARAMETER_NAME = "host-id";
+    public static final String IDENTIFIER_PARAMETER = "identifier";
+    public static final String HOST_ID_PARAMETER_NAME = "host-id";
+    public static final String SEARCH_PARAMETER = "search-parameter";
 
-    private static final String LUCENE_QUERY_WITH_HOST = "+parentpath:/application/themes/* +title:template.vtl +conhost:%s";
-    private static final String LUCENE_QUERY_WITH_OUT_HOST = "+parentpath:/application/themes/* +title:template.vtl";
+
+    private static final String BASE_LUCENE_QUERY         = "+parentpath:/application/themes/*";
+    private static final String LUCENE_QUERY_WITH_HOST    = "+parentpath:/application/themes/* +title:template.vtl +conhost:%s";
+    private static final String LUCENE_QUERY_WITHOUT_HOST = "+parentpath:/application/themes/* +title:template.vtl";
+
+    @VisibleForTesting
+    static final String THEME_PNG = "theme.png";
 
     private ContentletAPI contentletAPI;
 
+    private FolderAPI folderAPI;
+
     public ThemePaginator() {
-        this(APILocator.getContentletAPI());
+        this(APILocator.getContentletAPI(), APILocator.getFolderAPI());
     }
 
     @VisibleForTesting
-    ThemePaginator(final ContentletAPI contentletAPI) {
+    ThemePaginator(final ContentletAPI contentletAPI, FolderAPI folderAPI) {
         this.contentletAPI = contentletAPI;
+        this.folderAPI     = folderAPI;
     }
 
     @Override
-    public PaginatedArrayList<Contentlet> getItems(final User user, final int limit, final int offset,
+    public PaginatedArrayList<Folder> getItems(final User user, final int limit, final int offset,
                                                    final Map<String, Object> params) throws PaginationException {
 
         final OrderDirection direction =  params != null && params.get(ORDER_DIRECTION_PARAM_NAME) != null ?
                 (OrderDirection) params.get(ORDER_DIRECTION_PARAM_NAME) :
                 OrderDirection.ASC;
 
-        final String hostId = params != null ? (String) params.get(HOST_ID_PARAMETER_NAME) : null;
-        final String query = String.format(hostId == null ? LUCENE_QUERY_WITH_OUT_HOST : LUCENE_QUERY_WITH_HOST, hostId);
+        final String hostId       = params != null ? (String) params.get(HOST_ID_PARAMETER_NAME) : null;
```

Comment:
we should have something like 
final String hostId = UtilMethods.getOrNull (params, HOST_ID_PARAMETER_NAME);